server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java

```diff
@@ -129,17 +132,47 @@
    * avoid some tsfileResource is changed (e.g., from unsealed to sealed) when a query is executed.
    */
   private final ReadWriteLock closeQueryLock = new ReentrantReadWriteLock();
+  /**
+   * time range id in the storage group -> tsFileProcessor for this time range
+   */
+  private final HashMap<Long, TsFileProcessor> workSequenceTsFileProcessors = new HashMap<>();
+  /**
+   * time range id for divide storage group -> last used time of tsFileProcessor for this storage
+   * group
+   */
+  private final HashMap<Long, Long> sequenceTsfileProcessorLastUseTime = new HashMap<>();
+  /**
+   * time range id in the storage group -> tsFileProcessor for this time range
+   */
+  private final HashMap<Long, TsFileProcessor> workUnsequenceTsFileProcessors = new HashMap<>();
+  /**
+   * time range id for divide storage group -> last used time of tsFileProcessor for this storage
+   * group
+   */
+  private final HashMap<Long, Long> unsequenceTsfileProcessorLastUseTime = new HashMap<>();
+  // Time range for divide storage group, unit is second
+  private final long timeRangeForStorageGroup = IoTDBDescriptor.getInstance().getConfig()
+      .getTimeRangeForStorageGroup();
   /**
    * the schema of time series that belong this storage group
    */
   private Schema schema;
   // includes sealed and unsealed sequence TsFiles
-  private List<TsFileResource> sequenceFileList = new ArrayList<>();
-  private TsFileProcessor workSequenceTsFileProcessor = null;
+  private TreeSet<TsFileResource> sequenceFileList = new TreeSet<>(
+      new Comparator<TsFileResource>() {
+        @Override
+        public int compare(TsFileResource o1, TsFileResource o2) {
+          if (o1.getProcessor() == null || o2.getProcessor() == null) {
+            return compareFileName(o1.getFile(), o2.getFile());
```

Comment:
can we merge this comparison into compareFileName and rename to compareResource?

Reply:
Lots of file comparison are based on just file name rather than tsfile resource. So merge them may change a lot logic